platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
If you don't want `short` for events, does that not rule out a standard `poll(struct pollfd[], nfds_t, int)` in future?

`struct pollfd` contains shorts. Would we define a non-standard `struct pollfd`? We only had to invent a private `PollFH` here because of `FileHandle *` instead of `int`, but if we had integer file descriptors, we could have used a standard `struct pollfd`.

I'm still assuming there's a possibly of providing standard POSIX prototypes with integer file descriptors. Or is that itself a problem because they'd use `int fd`, not `filehandle_t fd`?

My general feeling is that if there is a standard type, you need to have a pretty solid reason for deviating from it. Doubly so if you might end up sharing libraries' existing defines for its values like `POLLIN`.  (Doesn't seem to happen with the current 3 toolchains, which don't have their own `<poll.h>`, but it's a possibility like the errno values).

I also note we're using `int` extensively for errno values rather than `errno_t`. If we want `poll_t`, we should probably change that, together with adding `filehandle_t`.

I see that C11 Annex K (but not C++11) does actually add `errno_t` for documentation purposes, but it must be a typedef for `int`. I guess if we similarly define that `poll_t` is always typedefed to `short` as part of the API, then we retain POSIX compatibility.